django/core/files/storage.py

```diff
@@ -140,24 +143,98 @@ def url(self, name):
         raise NotImplementedError('subclasses of Storage must provide a created_time() method')
 
     def modified_time(self, name):
         """
         Returns the last modified time (as datetime object) of the file
-        specified by name.
+        specified by name. Deprecated: use get_modified_time() instead.
         """
+        warnings.warn(
+            'Storage.modified_time() is deprecated in favor of get_modified_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
         raise NotImplementedError('subclasses of Storage must provide a modified_time() method')
 
+    def get_accessed_time(self, name):
+        """
+        Return the last accessed time (as a datetime object) of the file
+        specified by name. The datetime will be timezone-aware if USE_TZ=True.
+        """
+        # At the end of the deprecation:
+        # raise NotImplementedError('subclasses of Storage must provide a get_accessed_time() method')
+        warnings.warn(
+            'Storage.accessed_time() is deprecated. '
+            'Storage backends should implement get_accessed_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
+        dt = self.accessed_time(name)
+        return _possibly_make_aware(dt)
+
+    def get_created_time(self, name):
+        """
+        Return the creation time (as a datetime object) of the file
+        specified by name. The datetime will be timezone-aware if USE_TZ=True.
+        """
+        # At the end of the deprecation:
+        # raise NotImplementedError('subclasses of Storage must provide a get_created_time() method')
+        warnings.warn(
+            'Storage.created_time() is deprecated. '
+            'Storage backends should implement get_created_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
+        dt = self.created_time(name)
+        return _possibly_make_aware(dt)
+
+    def get_modified_time(self, name):
+        """
+        Return the last modified time (as a datetime object) of the file
+        specified by name. The datetime will be timezone-aware if USE_TZ=True.
+        """
+        # At the end of the deprecation:
+        # raise NotImplementedError('subclasses of Storage must provide a get_modified_time() method')
+        warnings.warn(
+            'Storage.modified_time() is deprecated. '
+            'Storage backends should implement get_modified_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
+        dt = self.modified_time(name)
+        return _possibly_make_aware(dt)
+
+
+def _possibly_make_aware(dt):
+    """
+    Convert a datetime object in the local timezone to aware
+    if USE_TZ is True.
+    """
+    # This function is only needed to help with the deprecations above and can
+    # be removed in Django 2.0, RemovedInDjango20Warning.
+    tz = timezone.get_default_timezone()
+    return timezone.make_aware(dt, tz)
```

Comment:
@aaugustin okay, this now almost works. The one remaining thing thrown up is that `_possibly_make_aware()` will, with `USE_TZ=True`, now return an aware object in system TZ, not in UTC. Converting from one TZ into another I'm not sure how to do _safely_. Is `dt.astimezone(timezone.utc)` safe in this context? The `datetime.datetime.astimezone` documentation suggest it probably is, but I'm never certain with timezones :-(

(Also, my comment about about not needing the `tz` argument is wrong because it uses `get_current_timezone()` not `get_default_timezone()`. But the tests caught that anyway.)